components/lib/componentbase/ComponentBase.js

```diff
@@ -519,8 +519,12 @@ export const ComponentBase = {
 
             const getPTClassValue = (...args) => {
                 const value = getOptionValue(...args);
-
-                return ObjectUtils.isString(value) ? { className: value } : value;
+                if (Array.isArray(value)) return { className: classNames(...value) };
+                if (ObjectUtils.isString(value)) return { className: value };
+                if (value?.hasOwnProperty('className') && Array.isArray(value.className)) {
+                    return { className: classNames(...value.className) };
+                }
+                return value;
```

Comment:
I thought you mentioned PrimeVue already has this but that is not what I am seeing?

See:  https://github.com/primefaces/primevue/blob/c1a78ef9222d0e7d3f353e13dcab43611c2ea884/components/lib/basecomponent/BaseComponent.vue#L129-L132